<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shane Warland" />


<title>SF3B1 project _coding start</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SF3B1 project _coding start</h1>
<h4 class="author">Shane Warland</h4>
<h4 class="date">7/14/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>leafcutter_ds_pipeline/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSwarlandleafcutterdspipelinetreea994d74397e20ff9766184049689e68991cf37adtargetblanka994d74a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Swarland/leafcutter_ds_pipeline/tree/a994d74397e20ff9766184049689e68991cf37ad" target="_blank">a994d74</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSwarlandleafcutterdspipelinetreea994d74397e20ff9766184049689e68991cf37adtargetblanka994d74a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Swarland/leafcutter_ds_pipeline/tree/a994d74397e20ff9766184049689e68991cf37ad" target="_blank">a994d74</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    code/.snakemake/
    Ignored:    code/clustering/
    Ignored:    code/logs/

Untracked files:
    Untracked:  analysis/Darmanhg38.xls
    Untracked:  analysis/leafcutter_perind.counts.denoms
    Untracked:  analysis/leafcutter_perind.counts.numers
    Untracked:  output/DifferentialSplicing/

Unstaged changes:
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SF3B1_RMarkdown_initialcode.Rmd</code>) and HTML (<code>docs/SF3B1_RMarkdown_initialcode.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Swarland/leafcutter_ds_pipeline/blob/a994d74397e20ff9766184049689e68991cf37ad/analysis/SF3B1_RMarkdown_initialcode.Rmd" target="_blank">a994d74</a>
</td>
<td>
Shane Warland
</td>
<td>
2020-07-23
</td>
<td>
wflow_publish(“SF3B1_RMarkdown_initialcode.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Swarland/leafcutter_ds_pipeline/blob/1e80761c15d80bf060b6ff2a14069a1a0b8f6085/analysis/SF3B1_RMarkdown_initialcode.Rmd" target="_blank">1e80761</a>
</td>
<td>
Shane Warland
</td>
<td>
2020-07-22
</td>
<td>
This is now a more complete pipeline
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Swarland/leafcutter_ds_pipeline/blob/a9a1fa002ab1ea8ccce84be16e451a3430db8a6f/analysis/SF3B1_RMarkdown_initialcode.Rmd" target="_blank">a9a1fa0</a>
</td>
<td>
Shane Warland
</td>
<td>
2020-07-22
</td>
<td>
Commiting these files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#Access necessary packages
library(data.table)
library(readxl)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)
library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    dcast, melt</code></pre>
<pre class="r"><code>library(ggrepel)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✔ tibble  2.1.3     ✔ purrr   0.3.4
✔ tidyr   1.1.0     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::between()   masks data.table::between()
✖ dplyr::filter()    masks stats::filter()
✖ dplyr::first()     masks data.table::first()
✖ dplyr::lag()       masks stats::lag()
✖ dplyr::last()      masks data.table::last()
✖ purrr::transpose() masks data.table::transpose()</code></pre>
<pre class="r"><code>library(gplots)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(knitr)
library(&quot;RColorBrewer&quot;)</code></pre>
<pre class="bash"><code>#comand for creating numerator and denominator files
#Specify the perind.counts.gz file from leafcutter clustering step
zcat /home/sfwarland/leafcutter_ds_pipeline/code/clustering/SF3B1K700E_v_WT/leafcutter_perind.counts.gz | perl -lne &#39;if ($.==1) {print} else {$_ =~ s/\d+\///g; print}&#39; &gt; leafcutter_perind.counts.numers
zcat /home/sfwarland/leafcutter_ds_pipeline/code/clustering/SF3B1K700E_v_WT/leafcutter_perind.counts.gz  | perl -lne &#39;if ($.==1) {print} else {$_ =~ s/\/\d+//g; print}&#39; &gt; leafcutter_perind.counts.denoms</code></pre>
<pre><code>gzip: /home/sfwarland/leafcutter_ds_pipeline/code/clustering/SF3B1K700E_v_WT/leafcutter_perind.counts.gz: No such file or directory
gzip: /home/sfwarland/leafcutter_ds_pipeline/code/clustering/SF3B1K700E_v_WT/leafcutter_perind.counts.gz: No such file or directory</code></pre>
<pre class="r"><code>#Assigning Number and Denominator Count table

Denoms &lt;- &#39;../output/CountTables/SF3B1K700E_v_WT.numers.gz&#39;
Numers &lt;- &#39;../output/CountTables/SF3B1K700E_v_WT.denoms.gz&#39;

#create the count tables
#If .gz add &quot;gunzip -c &quot; to paste0 function
NumeratorCountTable &lt;- fread(paste0(Numers), sep=&quot; &quot;, header=F, data.table=F)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;R.oo&#39;:
  method        from       
  throw.default R.methodsS3</code></pre>
<pre class="r"><code>DenominatorCountTable &lt;- fread(paste0(Denoms), sep=&quot; &quot;, header=F, data.table=F)
head(NumeratorCountTable)</code></pre>
<pre><code>                           V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11
1 chr2:219002:219747:clu_1_NA 10  0  0  3  7  7  0  2   0   0
2 chr2:219002:221464:clu_1_NA  0  0  0  1  1  3  0  0   0   0
3 chr2:219002:224864:clu_1_NA 56  0  0 19 39 22  0  9   0   7
4 chr2:219002:229966:clu_1_NA 82  0  0 65 29 66  3 45   8   0
5 chr2:224921:229966:clu_1_NA 71  0  0 19 47 34  0 13   0   8
6 chr2:230045:230547:clu_2_NA  2  0  0  3  4  7  1 11   0   0</code></pre>
<pre class="r"><code>#Create PSI_table and clean up
PSI_Table &lt;- NumeratorCountTable[-1]/DenominatorCountTable[-1]
head(PSI_Table)</code></pre>
<pre><code>   V2  V3  V4 V5 V6 V7  V8  V9 V10 V11
1   1 NaN NaN  1  1  1 NaN   1 NaN NaN
2 NaN NaN NaN  1  1  1 NaN NaN NaN NaN
3   1 NaN NaN  1  1  1 NaN   1 NaN   1
4   1 NaN NaN  1  1  1   1   1   1 NaN
5   1 NaN NaN  1  1  1 NaN   1 NaN   1
6   1 NaN NaN  1  1  1   1   1 NaN NaN</code></pre>
<pre class="r"><code>colnames(NumeratorCountTable)[1] &lt;- &quot;chrom&quot;
#gsub(&quot;:clu_.+&quot;, &quot;&quot;, NumeratorCountTable$chrom) %&gt;% head()
row.names(PSI_Table) &lt;- gsub(&quot;:clu_.+&quot;, &quot;&quot;, NumeratorCountTable$chrom)
PSI_Table[is.na(PSI_Table)] &lt;- 0

#Have to add column names for propper sorting later

col_names&lt;-fread(Numers, sep=&quot;&quot;, quote=F)[1,] %&gt;% colnames()
colnamessplit&lt;- unlist(str_split(col_names,&quot; &quot;))

data.table(colnamessplit)</code></pre>
<pre><code>    colnamessplit
 1:          9543
 2:         27168
 3:         14580
 4:         45109
 5:         30092
 6:         48124
 7:         32641
 8:         30090
 9:         36181
10:         28684</code></pre>
<pre class="r"><code>head(colnamessplit)</code></pre>
<pre><code>[1] &quot;9543&quot;  &quot;27168&quot; &quot;14580&quot; &quot;45109&quot; &quot;30092&quot; &quot;48124&quot;</code></pre>
<pre class="r"><code>colnames(PSI_Table)&lt;-colnamessplit

head(PSI_Table)</code></pre>
<pre><code>                   9543 27168 14580 45109 30092 48124 32641 30090 36181
chr2:219002:219747    1     0     0     1     1     1     0     1     0
chr2:219002:221464    0     0     0     1     1     1     0     0     0
chr2:219002:224864    1     0     0     1     1     1     0     1     0
chr2:219002:229966    1     0     0     1     1     1     1     1     1
chr2:224921:229966    1     0     0     1     1     1     0     1     0
chr2:230045:230547    1     0     0     1     1     1     1     1     0
                   28684
chr2:219002:219747     0
chr2:219002:221464     0
chr2:219002:224864     1
chr2:219002:229966     0
chr2:224921:229966     1
chr2:230045:230547     0</code></pre>
<p>Now lets filter the data based on significant introns identified in Darman et al.</p>
<pre class="r"><code>#Lets try filtering it by significant introns from Darman
Darman &lt;- read_excel(&quot;Darmanhg38.xls&quot;, col_names=T)

#used mutate to add 1 to both start and stop columns in Darman in order to clean up mapping 
Darman1 &lt;- Darman %&gt;% mutate(Start=Start+1,End=End+1) 
Darman1 %&gt;% head()</code></pre>
<pre><code># A tibble: 6 x 25
  Chromosome  Start    End Strand `Gene Symbols` `FDR q-value`
  &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;                  &lt;dbl&gt;
1 chr17      7.56e7 7.56e7 +      MYO15B         0.0222       
2 chrX       1.54e8 1.54e8 -      MECP2          0.000253     
3 chr12      1.16e8 1.16e8 -      MED13L         0.000000277  
4 chr4       2.07e7 2.07e7 +      PACRGL         0.000357     
5 chr10      5.46e6 5.46e6 +      NET1           0.000136     
6 chr7       1.02e8 1.02e8 +      ORAI2,PRKRIP1… 0.00000000801
# … with 19 more variables: `Alternative 3&#39; splice site` &lt;chr&gt;,
#   `Alternative 5&#39; splice site` &lt;chr&gt;, `Exon inclusion` &lt;chr&gt;, `Exon
#   skipping` &lt;chr&gt;, `Intron retention` &lt;chr&gt;, `Junction novelty` &lt;chr&gt;,
#   `NMD prediction` &lt;chr&gt;, `Significant in BRCA` &lt;chr&gt;, `Significant in
#   CLL` &lt;chr&gt;, `Significant in UVM` &lt;chr&gt;, `Significant in SKCM` &lt;chr&gt;,
#   `Average mutant PSI` &lt;dbl&gt;, `Average WT PSI` &lt;dbl&gt;, `PSI Log2 fold
#   change` &lt;dbl&gt;, `Ave mutant junction count` &lt;dbl&gt;, `Ave WT junction
#   count` &lt;dbl&gt;, `Canonical junction` &lt;chr&gt;, `Aberrant junction
#   sequence` &lt;chr&gt;, `Canonical junction sequence` &lt;chr&gt;</code></pre>
<pre class="r"><code>#combine Darman first 3 columns so it matches
Darman.df &lt;- unite(Darman1, Chrom, c(&quot;Chromosome&quot;,&quot;Start&quot;), sep = &quot;:&quot;, remove = TRUE, na.rm = FALSE)
head(Darman.df)</code></pre>
<pre><code># A tibble: 6 x 24
  Chrom    End Strand `Gene Symbols` `FDR q-value` `Alternative 3&#39;…
  &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;                  &lt;dbl&gt; &lt;chr&gt;           
1 chr1… 7.56e7 +      MYO15B         0.0222        True            
2 chrX… 1.54e8 -      MECP2          0.000253      False           
3 chr1… 1.16e8 -      MED13L         0.000000277   True            
4 chr4… 2.07e7 +      PACRGL         0.000357      False           
5 chr1… 5.46e6 +      NET1           0.000136      True            
6 chr7… 1.02e8 +      ORAI2,PRKRIP1… 0.00000000801 True            
# … with 18 more variables: `Alternative 5&#39; splice site` &lt;chr&gt;, `Exon
#   inclusion` &lt;chr&gt;, `Exon skipping` &lt;chr&gt;, `Intron retention` &lt;chr&gt;,
#   `Junction novelty` &lt;chr&gt;, `NMD prediction` &lt;chr&gt;, `Significant in
#   BRCA` &lt;chr&gt;, `Significant in CLL` &lt;chr&gt;, `Significant in UVM` &lt;chr&gt;,
#   `Significant in SKCM` &lt;chr&gt;, `Average mutant PSI` &lt;dbl&gt;, `Average WT
#   PSI` &lt;dbl&gt;, `PSI Log2 fold change` &lt;dbl&gt;, `Ave mutant junction
#   count` &lt;dbl&gt;, `Ave WT junction count` &lt;dbl&gt;, `Canonical
#   junction` &lt;chr&gt;, `Aberrant junction sequence` &lt;chr&gt;, `Canonical
#   junction sequence` &lt;chr&gt;</code></pre>
<pre class="r"><code>Darmansorted.df &lt;-  unite(Darman.df, Chrom, c(&quot;Chrom&quot;,&quot;End&quot;), sep = &quot;:&quot;, remove = TRUE, na.rm = FALSE)
  #paste(Darman$Chromosome, &quot;:&quot;, Darman$Start, &quot;:&quot;, Darman$End)
head(Darmansorted.df)</code></pre>
<pre><code># A tibble: 6 x 23
  Chrom Strand `Gene Symbols` `FDR q-value` `Alternative 3&#39;…
  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;                  &lt;dbl&gt; &lt;chr&gt;           
1 chr1… +      MYO15B         0.0222        True            
2 chrX… -      MECP2          0.000253      False           
3 chr1… -      MED13L         0.000000277   True            
4 chr4… +      PACRGL         0.000357      False           
5 chr1… +      NET1           0.000136      True            
6 chr7… +      ORAI2,PRKRIP1… 0.00000000801 True            
# … with 18 more variables: `Alternative 5&#39; splice site` &lt;chr&gt;, `Exon
#   inclusion` &lt;chr&gt;, `Exon skipping` &lt;chr&gt;, `Intron retention` &lt;chr&gt;,
#   `Junction novelty` &lt;chr&gt;, `NMD prediction` &lt;chr&gt;, `Significant in
#   BRCA` &lt;chr&gt;, `Significant in CLL` &lt;chr&gt;, `Significant in UVM` &lt;chr&gt;,
#   `Significant in SKCM` &lt;chr&gt;, `Average mutant PSI` &lt;dbl&gt;, `Average WT
#   PSI` &lt;dbl&gt;, `PSI Log2 fold change` &lt;dbl&gt;, `Ave mutant junction
#   count` &lt;dbl&gt;, `Ave WT junction count` &lt;dbl&gt;, `Canonical
#   junction` &lt;chr&gt;, `Aberrant junction sequence` &lt;chr&gt;, `Canonical
#   junction sequence` &lt;chr&gt;</code></pre>
<pre class="r"><code>#Make rowname actual column so I can filter it
PSI_TableRowNamed &lt;- cbind(Chrom = rownames(PSI_Table), PSI_Table)
#PSI_TableRowNamed &lt;- PSI_TableRowNamed[,-1]
head(PSI_TableRowNamed)</code></pre>
<pre><code>                                Chrom 9543 27168 14580 45109 30092 48124
chr2:219002:219747 chr2:219002:219747    1     0     0     1     1     1
chr2:219002:221464 chr2:219002:221464    0     0     0     1     1     1
chr2:219002:224864 chr2:219002:224864    1     0     0     1     1     1
chr2:219002:229966 chr2:219002:229966    1     0     0     1     1     1
chr2:224921:229966 chr2:224921:229966    1     0     0     1     1     1
chr2:230045:230547 chr2:230045:230547    1     0     0     1     1     1
                   32641 30090 36181 28684
chr2:219002:219747     0     1     0     0
chr2:219002:221464     0     0     0     0
chr2:219002:224864     0     1     0     1
chr2:219002:229966     1     1     1     0
chr2:224921:229966     0     1     0     1
chr2:230045:230547     1     1     0     0</code></pre>
<pre class="r"><code>#dat2 &lt;- data_qual[data_qual$ID %in% dat$ID, ]
DarmanFiltered.df &lt;- PSI_TableRowNamed[PSI_TableRowNamed$Chrom %in% Darmansorted.df$Chrom,]

dim(DarmanFiltered.df)</code></pre>
<pre><code>[1] 692  11</code></pre>
<pre class="r"><code>head(DarmanFiltered.df)</code></pre>
<pre><code>                                        Chrom 9543 27168 14580 45109 30092
chr2:24315246:24328052 chr2:24315246:24328052    0     1     0     1     0
chr2:26214577:26215053 chr2:26214577:26215053    1     1     0     1     1
chr2:27037703:27037815 chr2:27037703:27037815    1     1     0     1     1
chr2:27037893:27038146 chr2:27037893:27038146    1     1     0     1     1
chr2:27381722:27382126 chr2:27381722:27382126    1     1     0     1     1
chr2:46543189:46543754 chr2:46543189:46543754    1     1     1     1     1
                       48124 32641 30090 36181 28684
chr2:24315246:24328052     0     1     0     1     0
chr2:26214577:26215053     0     1     0     1     0
chr2:27037703:27037815     1     1     1     1     0
chr2:27037893:27038146     1     1     1     1     0
chr2:27381722:27382126     1     1     1     1     0
chr2:46543189:46543754     1     1     1     1     1</code></pre>
<pre class="r"><code>DarmanFinished &lt;- (DarmanFiltered.df %&gt;%data.matrix() %&gt;% as.data.frame() %&gt;% select(-Chrom)%&gt;% as.matrix())
head(DarmanFinished)</code></pre>
<pre><code>                       9543 27168 14580 45109 30092 48124 32641 30090
chr2:24315246:24328052    0     1     0     1     0     0     1     0
chr2:26214577:26215053    1     1     0     1     1     0     1     0
chr2:27037703:27037815    1     1     0     1     1     1     1     1
chr2:27037893:27038146    1     1     0     1     1     1     1     1
chr2:27381722:27382126    1     1     0     1     1     1     1     1
chr2:46543189:46543754    1     1     1     1     1     1     1     1
                       36181 28684
chr2:24315246:24328052     1     0
chr2:26214577:26215053     1     0
chr2:27037703:27037815     1     0
chr2:27037893:27038146     1     0
chr2:27381722:27382126     1     0
chr2:46543189:46543754     1     1</code></pre>
<p>Run the PCA</p>
<pre class="r"><code>#Running PC analysis
library(tidyverse)
DarmanFinished = scale(DarmanFinished)
PCResults &lt;- (DarmanFinished) %&gt;%
  #head(100) %&gt;%
  t() %&gt;%
  prcomp(center=T, scale=T)

PCResults.df &lt;- as.data.frame(PCResults$x[,1:10]) %&gt;% rownames_to_column()
head(PCResults.df)</code></pre>
<pre><code>  rowname          PC1        PC2         PC3        PC4        PC5
1    9543 -11.22568230   7.172131  -0.3968210  1.3921127 -0.5910683
2   27168  -0.02328187 -16.559457  -2.6660208 10.4858965 13.3559310
3   14580  30.64887497  11.800004 -17.0067299  0.1090665 -0.2137821
4   45109 -10.99079606   8.892463   0.2454178  1.0902339 -1.0231421
5   30092 -10.89631869   5.869837  -1.1482396  3.1755749 -3.3484961
6   48124 -10.16982643   7.891338   0.6027077 -3.0404127  3.9033717
          PC6        PC7          PC8          PC9         PC10
1  6.02000622 -7.0525179   5.08762550 -10.15755404 2.203099e-14
2 -0.05489443 -0.9144861  -0.54614203   0.91073190 1.174061e-14
3 -0.24362182 -0.1919491   0.08085549   0.03517435 1.765255e-14
4  0.87804289  1.3062776 -14.16656423  -1.78497788 1.624048e-14
5 10.92229172  5.6669894   3.95399441   7.95066222 1.081080e-14
6 -9.56932464 10.1401504   4.50780142  -3.45750517 1.676437e-14</code></pre>
<p>I have the PCR results but I want to add data from the group file in order to be able to cluster my PC plot by group This code processes it and adds it</p>
<pre class="r"><code>#getting rid of ID from group file 
groupfile &lt;- &quot;../data/groupfiles/sf3b1_k700e_wt.groups.tsv&quot;
groupfiletable &lt;- read.delim(groupfile, header = F, sep = &quot;\t&quot;)
head(groupfiletable)</code></pre>
<pre><code>       V1    V2
1 id27168 K700E
2 id32641 K700E
3 id36181 K700E
4 id28684    WT
5 id14580    WT
6 id30090    WT</code></pre>
<pre class="r"><code>groupNoID &lt;- data.frame(lapply(groupfiletable, function(x) {gsub(&quot;id&quot;, &quot;&quot;, x)}))

colnames(groupNoID) &lt;- c(&quot;rowname&quot;, &quot;mutation&quot;)
head(groupNoID,20)</code></pre>
<pre><code>   rowname mutation
1    27168    K700E
2    32641    K700E
3    36181    K700E
4    28684       WT
5    14580       WT
6    30090       WT
7    30092       WT
8     9543       WT
9    45109       WT
10   48124       WT</code></pre>
<pre class="r"><code>#merging group file
#Using new PCResults
combined.data &lt;- merge(x = PCResults.df, y = groupNoID, by = c(&quot;rowname&quot;), all = TRUE)
head(combined.data)</code></pre>
<pre><code>  rowname          PC1        PC2        PC3        PC4          PC5
1   14580  30.64887497  11.800004 -17.006730  0.1090665  -0.21378207
2   27168  -0.02328187 -16.559457  -2.666021 10.4858965  13.35593100
3   28684  21.80393877   2.852849  24.132634  0.2851606   0.05691744
4   30090 -10.46152440   7.472415   0.930207 -3.6107418   0.69629270
5   30092 -10.89631869   5.869837  -1.148240  3.1755749  -3.34849607
6   32641   0.23805147 -17.279302  -1.947562  6.1474118 -14.37692431
          PC6        PC7         PC8          PC9          PC10 mutation
1 -0.24362182 -0.1919491  0.08085549  0.035174349  1.765255e-14       WT
2 -0.05489443 -0.9144861 -0.54614203  0.910731904  1.174061e-14    K700E
3  1.01120054  0.1080250  0.16949782  0.009478321 -3.341771e-14       WT
4 -7.75225821 -9.8473433  1.41884014  7.914073597  3.053113e-15       WT
5 10.92229172  5.6669894  3.95399441  7.950662217  1.081080e-14       WT
6 -5.41974579  0.3532528  0.54966461 -1.056023519 -1.068590e-14    K700E</code></pre>
<p>Now visualize the PCA</p>
<pre class="r"><code>#PCA graph
PCResultsNoS2.df &lt;- PCResults.df[-c(5,6,11,12),]
ggplot(combined.data) +
  geom_point(aes(x=PC1, y=PC2, color= mutation))</code></pre>
<p><img src="figure/SF3B1_RMarkdown_initialcode.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#This was simply used to create bed files to trouble shoot
Darman %&gt;% select(&quot;Chromosome&quot;,&quot;Start&quot;,&quot;End&quot;) %&gt;% 
  write.table(&quot;/project2/yangili1/shanewarland/SF3B1_project/DarmanBed.bed&quot;,col.names =F, sep = &quot;\t&quot;, row.names=F, quote=F)

PSI_Table %&gt;% rownames_to_column() %&gt;% select(rowname) %&gt;% 
  separate(rowname, c(&quot;Chromosome&quot;,&quot;Start&quot;,&quot;End&quot;), sep=&quot;:&quot;)%&gt;% write.table(&quot;/project2/yangili1/shanewarland/SF3B1_project/PSI_BED.bed&quot;,col.names =F, sep = &quot;\t&quot;, row.names=F, quote=F)

&quot;chr6:10723241:10724556&quot; %in% PSI_TableRowNamed$Chrom</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code>arrange(Darmansorted.df, `FDR q-value`) %&gt;% head()</code></pre>
<pre><code># A tibble: 6 x 23
  Chrom Strand `Gene Symbols` `FDR q-value` `Alternative 3&#39;…
  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;                  &lt;dbl&gt; &lt;chr&gt;           
1 chr6… +      SYCP2L,TMEM14…      2.09e-31 True            
2 chr1… +      DYNLL1              7.02e-29 True            
3 chr1… +      ZDHHC16             5.93e-23 True            
4 chr2… +      TCEA2               7.31e-23 True            
5 chr1… -      SERBP1              3.64e-22 True            
6 chr1… +      ZNF771              2.05e-21 True            
# … with 18 more variables: `Alternative 5&#39; splice site` &lt;chr&gt;, `Exon
#   inclusion` &lt;chr&gt;, `Exon skipping` &lt;chr&gt;, `Intron retention` &lt;chr&gt;,
#   `Junction novelty` &lt;chr&gt;, `NMD prediction` &lt;chr&gt;, `Significant in
#   BRCA` &lt;chr&gt;, `Significant in CLL` &lt;chr&gt;, `Significant in UVM` &lt;chr&gt;,
#   `Significant in SKCM` &lt;chr&gt;, `Average mutant PSI` &lt;dbl&gt;, `Average WT
#   PSI` &lt;dbl&gt;, `PSI Log2 fold change` &lt;dbl&gt;, `Ave mutant junction
#   count` &lt;dbl&gt;, `Ave WT junction count` &lt;dbl&gt;, `Canonical
#   junction` &lt;chr&gt;, `Aberrant junction sequence` &lt;chr&gt;, `Canonical
#   junction sequence` &lt;chr&gt;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2 knitr_1.23         gplots_3.0.1.1    
 [4] forcats_0.4.0      stringr_1.4.0      purrr_0.3.4       
 [7] readr_1.3.1        tidyr_1.1.0        tibble_2.1.3      
[10] tidyverse_1.3.0    ggrepel_0.8.1      reshape2_1.4.3    
[13] ggplot2_3.2.1      dplyr_1.0.0        readxl_1.3.1      
[16] data.table_1.12.8 

loaded via a namespace (and not attached):
 [1] httr_1.4.1         jsonlite_1.6       R.utils_2.9.0     
 [4] modelr_0.1.8       gtools_3.8.1       assertthat_0.2.1  
 [7] cellranger_1.1.0   yaml_2.2.0         pillar_1.4.2      
[10] backports_1.1.4    lattice_0.20-38    glue_1.4.1        
[13] digest_0.6.20      promises_1.0.1     rvest_0.3.5       
[16] colorspace_1.4-1   htmltools_0.3.6    httpuv_1.5.1      
[19] R.oo_1.22.0        plyr_1.8.4         pkgconfig_2.0.2   
[22] broom_0.5.2        haven_2.3.1        scales_1.1.0      
[25] gdata_2.18.0       whisker_0.3-2      later_0.8.0       
[28] git2r_0.26.1       generics_0.0.2     farver_2.0.1      
[31] ellipsis_0.2.0.1   withr_2.1.2        lazyeval_0.2.2    
[34] cli_1.1.0          magrittr_1.5       crayon_1.3.4      
[37] evaluate_0.14      R.methodsS3_1.7.1  fs_1.3.1          
[40] fansi_0.4.0        nlme_3.1-140       xml2_1.3.2        
[43] tools_3.6.1        hms_0.5.3          lifecycle_0.2.0   
[46] munsell_0.5.0      reprex_0.3.0       compiler_3.6.1    
[49] caTools_1.17.1.2   rlang_0.4.6        grid_3.6.1        
[52] rstudioapi_0.10    bitops_1.0-6       labeling_0.3      
[55] rmarkdown_1.13     gtable_0.3.0       DBI_1.1.0         
[58] R6_2.4.0           lubridate_1.7.4    utf8_1.1.4        
[61] workflowr_1.6.2    rprojroot_1.3-2    KernSmooth_2.23-15
[64] stringi_1.4.3      Rcpp_1.0.3         vctrs_0.3.1       
[67] dbplyr_1.4.2       tidyselect_1.1.0   xfun_0.8          </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
